core/Color.carp

```diff
@@ -26,7 +26,7 @@
      BgWhite)
   (use Id)
   (defn hash [k]
-    (get-tag k))
+    (get-tag (the (Ref Color.Id) k)))
```

Comment:
So the error was a bit subtle, here was the problem:

1. When we qualify a function, we create two new environments, a function env and a recursive env, respectively, the parent of the recursive env is the original inner env from the context (e.g. current module being evaluated)
2. We then pass this function env to qualify syms, however, I didn't copy over the *Used* modules of the original env.

So the solution is to copy the use'd envs from the module over to the function env. 

Similarly, I noticed a but in primitiveUse that I fixed, and updated primitive use to take account of the context (for this particular case, color `use`s `Id`, but really it means to use `Color.Id`, so Use now accounts for this.

Reply:
The reason it was failing is because `using` ID fixes the type correctly via `get-tag`, but since it originally wasn't looking at the use modules correctly, it was falling through to `lookupeverywhere` and keeping the function generic.

Reply:
But, I'll have to think about the right approach a bit more, as this introduces some other problematic cases (e.g. `copy`, etc. We probably need to introduce an interface check).

Reply:
Fixed. I actually just made a mistake. All we needed to do was copy over use module paths to function envs.